hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -164,97 +169,66 @@ case class MatrixNativeReader(path: String) extends MatrixReader {
 
   def fullType: MatrixType = spec.matrix_type
 
-  def apply(mr: MatrixRead): MatrixValue = {
+  override def canLower: Boolean = true
+
+  override def lower(mr: MatrixRead): TableIR = {
+    val rowsPath = path + "/rows"
+    val entriesPath = path + "/entries"
+    val colsPath = path + "/cols"
+
     val hc = HailContext.get
 
-    val requestedType = mr.typ
-    assert(PruneDeadFields.isSupertype(requestedType, spec.matrix_type),
-      s"R: ${ requestedType.parsableString() }\nS: ${ spec.matrix_type.parsableString() }")
+    var tr: TableIR = TableRead(
+      TableType(
```

Comment:
The globals are stored in their own file, right? I don't see them being read in anywhere? (Do we have tests for this?)

Reply:
(discussed, but for other prying eyes: the rows/cols/entries all have relative links to the globals and are read in with the globals)